pkg/statistics/handle/bootstrap.go

```diff
@@ -68,21 +68,30 @@ func (*Handle) initStatsMeta4Chunk(cache statstypes.StatsCache, iter *chunk.Iter
 		physicalID = row.GetInt64(1)
 		maxPhysicalID = max(physicalID, maxPhysicalID)
 		newHistColl := *statistics.NewHistColl(physicalID, row.GetInt64(3), row.GetInt64(2), 4, 4)
+		// During the initialization phase, we need to initialize LastAnalyzeVersion with the snapshot,
+		// which ensures that we don't duplicate the auto-analyze of a particular type of table.
+		// When the predicate columns feature is turned on, if a table has neither predicate columns nor indexes,
+		// then auto-analyze will only analyze the _row_id and refresh stats_meta,
+		// but since we don't have any histograms or topn's created for _row_id at the moment.
+		// So if we don't initialize LastAnalyzeVersion with the snapshot here,
+		// it will stay at 0 and auto-analyze won't be able to detect that the table has been analyzed.
+		// But in the future, we maybe will create some records for _row_id, see:
+		// https://github.com/pingcap/tidb/issues/51098
 		snapshot := row.GetUint64(4)
+		lastAnalyzeVersion, lastStatsFullUpdateVersion := snapshot, snapshot
+		if !row.IsNull(5) {
+			lastAnalyzeVersion = max(lastAnalyzeVersion, row.GetUint64(5))
+			lastStatsFullUpdateVersion = max(lastStatsFullUpdateVersion, row.GetUint64(5))
```

Comment:
This would be an intest.Assert. Because normally it will always be greater than or equal to lastAnalyzeVersion.